test/functional/test_framework/util.py

```diff
@@ -408,13 +416,22 @@ def sync_blocks(rpc_connections, *, wait=1, timeout=60):
         if best_hash.count(best_hash[0]) == len(rpc_connections):
             return
         # Check that each peer has at least one connection
-        assert (all([len(x.getpeerinfo()) for x in rpc_connections]))
+        assert (all([len(x.getpeerinfo()) for x in get_nodes_least_version(rpc_connections)]))
```

Comment:
I don't like making sync_**all** only sync the latest versions. Why is this needed here anyway?

Reply:
Because the v0.15.2 node has no peers. But you're right, if I am not using *sync_blocks* it should not react at all. In a previous version of upgradewallet test there was a *sync_blocks* call. Will adapt the code.